### ISSUE TITLE
[Bug] Iron golem AI seems inconsistent

### ISSUE THREAD

**Issue by maxanier** [TeamLapen/Vampirism#856]
## Versions

- Minecraft: 1.16.4
- Forge: 
- Vampirism: 1.7.5

## Issue Description
Summoned golems work fine, while naturally spawned golems supposedly act weird

## Reproduce Steps
Unsure. Might occur in naturally generated vampire villages.

**Comment by Cheaterpaul** [comment]
I have tested it several times with different situations


- (vampire village ->) hunter village -> vampire village
- neutral village -> vampire village
both with normal capture and force capture.

But everything seems to behave normal.
I tried to have the golem attack vampires until the capture was complete since it was the only thing i could think of, but expected behavior.

The last thing i could think of is that the modified target goal
https://github.com/TeamLapen/Vampirism/blob/57d6f756c47c56967caa5540167b03fbbd2ddd13/src/main/java/de/teamlapen/vampirism/entity/ModEntityEventHandler.java#L255
is either not added/removed or another targetgoal is added with higher priority. (all probably caused by another mod)
